src/gui/userstatusselectormodel.h

```diff
@@ -80,52 +91,43 @@ class UserStatusSelectorModel : public QObject
     Q_INVOKABLE void setUserStatus();
     Q_INVOKABLE void clearUserStatus();
 
-    Q_REQUIRED_RESULT int predefinedStatusesCount() const;
-    Q_INVOKABLE UserStatus predefinedStatus(int index) const;
-    Q_INVOKABLE QString predefinedStatusClearAt(int index) const;
-    Q_INVOKABLE void setPredefinedStatus(int index);
+    QVector<UserStatus> predefinedStatuses() const;
 
-    Q_REQUIRED_RESULT QStringList clearAtValues() const;
-    Q_REQUIRED_RESULT QString clearAt() const;
-    Q_INVOKABLE void setClearAt(int index);
+    Q_INVOKABLE void setPredefinedStatus(const UserStatus &predefinedStatus);
```

Comment:
same question about being a slot instead of an invokable method

Reply:
Turned into a slot